ljmd.py

```diff
@@ -1,5 +1,169 @@
+
+		self.natoms = int(data[0])
+		self.mass = float(data[1])
+		self.epsilon = float(data[2])
+		self.sigma = float(data[3])
+		self.rcut = float(data[4])
+		self.box = float(data[5])
+		self.restfile = data[6]
+		self.trajfile = data[7]
+		self.ergfile = data[8]
+		self.nsteps = int(data[9])
+		self.dt = float(data[10])
+		self.nprint = int(data[11])
+
+		self.sys = MDSYS_T(natoms=self.natoms,
+							mass=self.mass,
+							epsilon=self.epsilon,
+							sigma=self.sigma,
+							rcut=self.rcut,
+							box=self.box,
+							nsteps = self.nsteps,
+							dt=self.dt)
+
+		self.sys.rx = (c_double*self.natoms)()
+		self.sys.ry = (c_double*self.natoms)()
+		self.sys.rz = (c_double*self.natoms)()
+		self.sys.vx = (c_double*self.natoms)()
+		self.sys.vy = (c_double*self.natoms)()
+		self.sys.vz = (c_double*self.natoms)()
+		self.sys.fx = (c_double*self.natoms)()
+		self.sys.fy = (c_double*self.natoms)()
+		self.sys.fz = (c_double*self.natoms)()
+
+		self.restart("examples/"+self.restfile)
+
+		self.force()
+		self.ekin()
+		self.sys.nfi = 0
+
+	def run_simulation(self):
+		print("Running simulation")
+
+		self.erg = open(self.ergfile, 'w')
+		self.traj = open(self.trajfile, 'w')
+		self.write_output()
+		self.sys.nfi = 1
+		while self.sys.nfi <= self.sys.nsteps:
+			if (self.sys.nfi%self.nprint)==0:
+				self.write_output()
+
+			self.propagate1()
+			self.force()
+			self.propagate2()
+			self._ljmd.ekin(byref(self.sys))
+			self.sys.nfi+=1
+			
+		print("Done simulation")
+		
+		self.erg.close()
+		self.traj.close()
+	
+	def propagate1(self):
+		for i in range(self.natoms):
+			self._ljmd.propagate_velocity(byref(self.sys), i)
+			self._ljmd.propagate_position(byref(self.sys), i)
+
+	def force(self):
+		self._ljmd.force(byref(self.sys))
+
+	def propagate2(self):
+		for i in range(self.natoms):
+			self._ljmd.propagate_velocity(byref(self.sys), i)
+
+	def ekin(self):
+		print("Kinetic energy initialized")
+		self._ljmd.ekin(byref(self.sys))
+	
+	def write_output(self):
+		self.erg.write("% 8d % 20.8f % 20.8f % 20.8f % 20.8f\n" % (self.sys.nfi, self.sys.temp,
+         				self.sys.ekin, self.sys.epot, self.sys.ekin + self.sys.epot))
+
+		self.traj.write("%d\n nfi=%d etot=%20.8f\n" % (self.sys.natoms, self.sys.nfi, 
+						self.sys.ekin + self.sys.epot))
+
+		for i in range(self.natoms):
+			self.traj.write("Ar  %20.8f %20.8f %20.8f\n" % (self.sys.rx[i], self.sys.ry[i],
+         		   			self.sys.rz[i]))
+
+	def read_input(self, datafile):
+		try:
+			with open(datafile) as f:
+				data = [x.split()[0] for x in f.readlines()] 
+		except FileNotFoundError:
+			print("Missing input file: Please provide correct file path")
+			raise
+
+		assert(len(data)==12), "Invalid input file: File should have 12 rows of input values"
+
+		return data
+	
+	def restart(self, restfile):
+		with open(restfile) as f:
+			r = f.readlines()
+			assert(len(r)==2*self.natoms), "Invalid restart file: File should have rows equal to twice the number of atoms\n"
```

Comment:
is this line OK?

Reply:
Yes except for the format. I'll look on how to automate the formatting.

Reply:
Got it